File: listjobs_ipython.py
# -*- coding: utf-8 -*-

import string
import DrQueue
from DrQueue import Job as DrQueueJob
from DrQueue import Client as DrQueueClient


def main():
    # initialize DrQueue client
    client = DrQueueClient()
    # fetch a list of all jobs
    jobs = client.query_all_jobs()
        
    # walk through tasks of every job
    for job in jobs:
        jmsg_id = job['msg_id']
        jheader = job['header']
        tasks = []
        
        print("\nTasks of job %s:" % jmsg_id)
        print("msg_id                                 status    owner       completed at")
        
        for task_id in jheader['after']:
            task = client.query_task(task_id)
            
            tmsg_id = task['msg_id']
            theader = task['header']
            username = theader['username']
        
            if task['completed'] == None:
                status = "pending"
            else:
                result_header = task['result_header']
                status = result_header['status']
                cpl = task['completed']
                result_content = eval(task['result_content'])
                result_buffers = eval(task['result_content'])
                buffers = task['buffers']
                pyerr = task['pyout']
    
            print("%s   %s  %s  %i-%02i-%02i %02i:%02i:%02i" % (tmsg_id, string.ljust(status, 8), string.ljust(username, 10), cpl.year, cpl.month, cpl.day, cpl.hour, cpl.minute, cpl.second))
        #print(result_content)
        #print(result_buffers)
        #print(buffers)
        #print(pyerr)
    
if __name__ == "__main__":
    main()

File: listjobs_ipython.rb
require "rubypython" 
RubyPython.start(:python_exe => "python2.6")
sys = RubyPython.import "sys" 
sys.argv = [""]
DrQueue = RubyPython.import("DrQueue")


# initialize DrQueue client
client = DrQueue.Client.new
# fetch a list of all jobs
jobs = client.query_job_list()

jobs.to_enum.each do |job|
  tasks = client.query_task_list(job)
  
  puts("\nTasks of job "+job.to_s+":")
  puts("msg_id                                 status    owner       completed at")

  # to_enum provides Ruby-like iteration
  tasks.to_enum.each do |task|
    tmsg_id = task['msg_id']
    theader = task['header']
    username = theader['username']

    if task['completed'] == nil
      status = "pending"
    else
      result_header = task['result_header']
      status = result_header['status']
      cpl = task['completed']
    end
    # converting Python string to Ruby string and then to Ruby integer helps for using format strings
    puts(tmsg_id+"   "+status.ljust(8)+"  "+username.ljust(10)+"  "+cpl.year.to_s+"-"+("%02i-%02i %02i:%02i:%02i" % [cpl.month.to_s.to_i, cpl.day.to_s.to_i, cpl.hour.to_s.to_i, cpl.minute.to_s.to_i, cpl.second.to_s.to_i]))
  end
end


RubyPython.stop
